### DISCUSSION TITLE
Add new node Foreach

### DISCUSSION THREAD

**Discussion by sohelzerdoumi** [langgenius/dify#3463] [Suggestion]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] Pleas do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing?

Given a long input text, I would like to handle the blocks independently, then merge the results.

For exemple, summerize very long text

### 2. Describe the feature you'd like to see

I would like 2 nodes:
* Foreach node: based on array input, it will handle next flow independently
* Merge node: based on previous flow, it will end Foreach, then assign all outputs to single array variable

### 3. How will this feature improve your workflow or experience?

This can solve my challenge

### 5. Can you help us with this feature?

- [ ] I am interested in contributing to this feature.

**Comment by crazywoola** [comment]
See https://docs.dify.ai/features/workflow/key-concept It's in our roadmap. Thanks for the suggestion.

**Reply by sohelzerdoumi** [reply]
Oh, my bad
thx for your answer !
